### ISSUE TITLE
Bug: no-require-imports applies to `.js` files in CommonJS projects, and to `.cjs` files

### ISSUE THREAD

**Issue by simonhaenisch** [typescript-eslint/typescript-eslint#9730]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.5.2&fileType=.cjs&code=MYewdgzgLgBAZhGBeGAnApgRwK4EsMAUA5AkQJQBQFokIANugHR0gDmBClQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1ukQEdZKPWpQC2xDtHyoMkRNGgTI4MAF8QKoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
const fs = require('fs')

console.log(fs)
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-require-imports": "error",
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // doesn't matter
  }
}
```

### Expected Result

In a `.cjs` file, `no-require-imports` should not report any errors since the file extension enforces that the module is written as CommonJS.

In a `.js` file, if `package.json` doesn't set `"type": "module"`, the same should apply.

### Actual Result

It reports errors for `require()` calls even in `.cjs` files, but also in `.js` files when the package isn't set to `"type": "module"`.

### Additional Info

ESLint 9 does check the package type and file extensions to set `languageOptions.sourceType` automatically. Maybe that info can be used by the rule?

**Comment by bradzacher** [comment]
The rule is intended for TS files - not JS files.
In TS files you use ESM syntax regardless of target and using `require` is untyped, unsafe and bad.

It wasn't really intended for JS/CJS/MJS files where the situation is vastly different.

We generally avoid adding specific handling for file extensions. We used to do it but we found it can end up breaking certain setups.

Instead our advice is to disable rules using your eslint config.
https://eslint.org/docs/latest/use/configure/rules#using-configuration-files-1
